test/unit/services/query-service.test.js

```diff
@@ -119,5 +119,214 @@ describe('query-service', () => {
       expect(validatedResponseNoContext).toStrictEqual(false)
       expect(validatedResponseNoAnswer).toStrictEqual(false)
     })
+
+    test('fetchAnswer returns summary answer when summariesEnabled is true', async () => {
+      createRetrievalChain.mockResolvedValue({
+        invoke: jest.fn().mockResolvedValue({
+          answer: JSON.stringify({ answer: 'summary response', items: [] })
+        })
+      })
+
+      const prompt = jest.fn()
+      const input = 'deer fencing'
+      ChatPromptTemplate.fromMessages.mockReturnValue(prompt)
+
+      const response = await fetchAnswer({}, input, [], false, true)
+
+      expect(JSON.parse(response).answer).toStrictEqual('summary response')
```

Comment:
`runFetchAnswerQuery` could return summariesMode back to the caller. So you can expect it here
`expect(JSON.parse(response).summariesMode).toBe(true)`